pkg/probe/http/http.go

```diff
@@ -94,10 +94,10 @@ func DoHTTPProbe(url *url.URL, headers http.Header, client GetHTTPInterface) (pr
 		// Convert errors into failures to catch timeouts.
 		return probe.Failure, err.Error(), nil
 	}
+	if headers == nil {
```

Comment:
it is not really necessary as the if statement will work without issue even when `headers` is `nil`.

Reply:
this is on the edge of being a trivial unnecessary edit. I agree the code is better this way so it's lgtm

/lgtm
/hold cancel